app/assets/javascripts/initializers/initializePodcastPlayback.js

```diff
@@ -183,9 +183,14 @@ function initializePodcastPlayback() {
     Array.prototype.forEach.call(records, function (record) {
       var episodeSlug = record.getAttribute('data-episode');
       var podcastSlug = record.getAttribute('data-podcast');
-      record.onclick = function () {
+
+      var togglePodcastState = function (e) {
+        var isPressed = record.getAttribute('aria-pressed') === 'true';
+        e.preventDefault();
         if (podcastBarAlreadyExistAndPlayingTargetEpisode(episodeSlug)) {
           var audio = getById('audio');
+          record.setAttribute('aria-pressed', !isPressed);
```

Comment:
For the most part this is working really well, but I noticed it's possible to get into a bad state, e.g:

It's difficult to see in the video, but the `aria-pressed` state has ended up the opposite of what it should be somehow.


https://user-images.githubusercontent.com/20773163/126169279-7869e56e-cd5d-4fcc-8657-6735db30b3f4.mp4

I think it's because we rely on toggling to the opposite of the previous `isPressed` value, when there's a bit more going on in this code - sometimes we play or pause the current podcast, and sometimes we interrupt an already playing podcast with a new one (so aria-pressed should stay true instead of flipping to false).

I think maybe the most solid approach would be to let `playPause(audio)` handle setting `aria-pressed` for the case where we're already playing the episode, and then in the `else` block here where we call `loadAndPlayNewPodcast(episodeSlug)`, we also set `aria-pressed` to `true` since we're 'pressing play'.

What do you think?

Reply:
Nice catch. Fixing it this way, would leave out the case when the playback is controlled via the bottom bar (since the change is attached to the onClick handler). I abstracted it out to be in sync with the playback; no matter where it's controlled from (how we handle the addition/elimination of `playing` class for this button).

Please check the latest commit; also should we abstract this change to a separate function due to the duplication happening? (seemed like a minor thing to make a separate function for)

Also, I had a question. Why do we have separate functions for handling play/pause states?